src/posts/introducing-inference-data/index.md

```diff
@@ -0,0 +1,114 @@
+the Probabilistic Programming Language (PPL) Designer, and what things used to be like before `az.InferenceData`.
+
+### The Modern Bayesian Practitioner
+
+Like most modern statisticians, the Modern Bayesian Statistician uses a computer to perform their work.
+Many PPLs are available for this task, such as [Stan](https://mc-stan.org/),
+[PyMC](https://www.pymc.io/), [Tensorflow Probability](https://www.tensorflow.org/probability), [NumPyro](http://pyro.ai/numpyro/) etc.
+
+With probabilistic programming language in hand, the Bayesian then follows the "Bayesian workflow" below, performing each of the various steps
+as part of their analysis.
+
+![Bayesian Workflow](https://bayesiancomputationbook.com/_images/Bayesian_workflow.png)
+
+At each of these steps the PPLs generate many arrays, and not only single dimension arrays, but multi dimensional arrays.
+The prior predictive distribution, the posterior distribution, the posterior predictive distribution, the log likelihood, and even the input data
+are all numerical quantities that need to be captured and ordered to facilitate analysis.
+
+Often these arrays need to be passed to specialized functions in order to check for Markov Chain Monte Carlo (MCMC) convergence, model comparison, or getting summaries from the target distribution.
+
+And once the individual statistician has completed their work, they may either want to save it to disk,
+or share it with a colleague.
+
+## The Modern PPL designer
+
+Now consider a PPL designer. PPL designers typically want to focus on model building and inference: graph representation, samplers, symbolic differentiation speed and stability.
+However for their PPL to be useful they must ensure that users have easy access to diagnostics,
+visualizations, and model criticism tools.
+Suddenly the scope of their codebase grows, and duplicative functionality now exists across
+the open source Bayesian ecosystem.
+
+## Life before az.InferenceData
+
+Before `az.InferenceData` there wasn't a consistent object or method both for the user or for the PPL library maintainers.
+Things like posteriors, prior predictives, observed data, and sampler stats could be stored in their own ways,
+For example one PPL may store their modeling outputs in a dictionary, other may have used plain numpy arrays, a third had its own custom class.
+And that's even before considering the shapes and indices of the array.
+
+This all meant
+
+- Each PPL designer would have to implement their own internal objects
+  - The objects may not be serializable to disk
+- All adjacent workflow libraries would only work with a subset of these objects, or would require users to restructure data each time.
+- The APIs and interfaces for all of these were inconsistent making life challenging for end users
+  - Folks using different PPLs would be isolated in completely different ecosystems, even when the PPLs share the same general programming language.
```

Comment:
Just some light restructuring to avoid the nested list:

```suggestion
- Each PPL designer would have to implement their own data objects and would have to solve lots problems like how to serialize data to disk.
- All adjacent workflow libraries would only work with a subset of these objects, or would require users to restructure data each time.
- The APIs and interfaces for all of these were inconsistent making life challenging for end user. Users of different PPLs would be isolated in completely different ecosystems, even when the PPLs share the same general programming language.
```